packages/openapi-generator/src/file-serializer/index-file.spec.ts

```diff
@@ -1,39 +1,64 @@
+  readPrettierConfig
+} from '@sap-cloud-sdk/generator-common/internal';
 import { OpenApiDocument } from '../openapi-types';
 import { apiIndexFile, schemaIndexFile } from './index-file';
 
-it('apiIndexFile serializes the api index file with referenced schemas', () => {
-  const document = {
-    apis: [{ name: 'TestApi' }, { name: 'DefaultApi' }],
-    schemas: [{}]
-  } as OpenApiDocument;
-  expect(apiIndexFile(document)).toMatchInlineSnapshot(`
-    "    export * from './test-api';
-        export * from './default-api';
-        export * from './schema';"
-  `);
-});
+describe('index-file', () => {
+  it('apiIndexFile serializes the api index file with referenced schemas', () => {
+    const document = {
+      apis: [{ name: 'TestApi' }, { name: 'DefaultApi' }],
+      schemas: [{}]
+    } as OpenApiDocument;
+    expect(apiIndexFile(document)).toMatchSnapshot();
+  });
 
-it('apiIndexFile serializes the api index file without referenced schemas', () => {
-  const document = {
-    apis: [{ name: 'TestApi' }, { name: 'DefaultApi' }],
-    schemas: []
-  } as unknown as OpenApiDocument;
-  expect(apiIndexFile(document)).toMatchInlineSnapshot(`
-    "    export * from './test-api';
-        export * from './default-api';"
-  `);
-});
+  it('apiIndexFile serializes the api index file following the esm format', async () => {
+    const document = {
+      apis: [{ name: 'TestApi' }, { name: 'DefaultApi' }],
+      schemas: [{}]
+    } as OpenApiDocument;
+    const createFileOptions: CreateFileOptions = {
+      generateESM: true,
+      overwrite: false,
+      prettierOptions: await readPrettierConfig(undefined)
+    };
+    expect(apiIndexFile(document, createFileOptions)).toMatchSnapshot();
+  });
+
+  it('apiIndexFile serializes the api index file without referenced schemas', () => {
+    const document = {
+      apis: [{ name: 'TestApi' }, { name: 'DefaultApi' }],
+      schemas: []
+    } as unknown as OpenApiDocument;
+    expect(apiIndexFile(document)).toMatchSnapshot();
+  });
+
+  it('schemaIndexFile serializes the schema index file for schemas in a document', () => {
+    const document = {
+      apis: [],
+      schemas: [
+        { schemaName: 'MySchema1', fileName: 'my-schema-1' },
+        { schemaName: 'MySchema2', fileName: 'some-other-name' }
+      ]
+    } as unknown as OpenApiDocument;
+    expect(schemaIndexFile(document)).toMatchSnapshot();
+  });
 
-it('schemaIndexFile serializes the schema index file for schemas in a document', () => {
-  const document = {
-    apis: [],
-    schemas: [
-      { schemaName: 'MySchema1', fileName: 'my-schema-1' },
-      { schemaName: 'MySchema2', fileName: 'some-other-name' }
-    ]
-  } as unknown as OpenApiDocument;
-  expect(schemaIndexFile(document)).toMatchInlineSnapshot(`
-    "export * from './my-schema-1';
-    export * from './some-other-name';"
-  `);
+  it('schemaIndexFile serializes the schema index file for schemas following the ESM format', async () => {
```

Comment:
I know this is old code, but seems we didn't follow the convention in most places